principles.md

```diff
@@ -3,12 +3,12 @@
 <br>
 
 ### Proposal Principles to Use to Achieve the Goal​
-- Boy Scout Rule​
+~~- Boy Scout Rule​~~ this is just a minor improvement (may risk of breaking things)
```

Comment:
Hey @buibaoanh,
I found it very efficient when we do small improvement consistently. Benefits are:
- Scope of your ticket actually does not change much.
- Done this one time, you don't see much different. But 10 times, the improvements are significant already.
- We should not afraid of breaking things. One trick could help is: have proper unit tests for a function before you change. Then, unit test could be sth you should consider to improve first.

Reply:
Agree on improving unit test first, after that we can talk about this `Boy Scout Rule`

Reply:
@theanh @sandyandi , can we enforce unit tests for new features, given legacy code does not have enough code coverage? think of unit tests as documentation rather than a burden

Reply:
1. I think we do have CI to check unit test coverage for frontend.
2. With test coverage is not enough, having proper test is sth we can not automate yet.
    In this case, we would you all help when you do review for a PR. Please also review the unit test. From there, we could  have better unit test to properly cover our code.
    
 Bao Anh, do you think it would help?

Reply:
Unit test coverage won't make sure unit tests are well written.
Reviewing unit test code is one way to achieve this, I am not sure if we can spend time to do this?

Reply:
> Unit test coverage won't make sure unit tests are well written.

Exactly.

> Reviewing unit test code is one way to achieve this, I am not sure if we can spend time to do this?

Why not, you should review unit tests, too.